graph/src/blockchain/block_stream.rs

```diff
@@ -1,21 +1,57 @@
-use super::{BlockPtr, Blockchain};
 use anyhow::Error;
 use async_trait::async_trait;
-use futures03::Stream;
+use futures::Stream;
+use std::cmp;
+use std::collections::VecDeque;
+use std::mem;
+use std::time::Duration;
+
+use crate::components::store::BlockNumber;
+use crate::components::store::WritableStore;
+use crate::{prelude::*, prometheus::labels};
+
+use super::{Block, BlockPtr, Blockchain};
+
+#[cfg(debug_assertions)]
+use fail::fail_point;
 
 pub struct BlockWithTriggers<C: Blockchain> {
```

Comment:
I'd really like to see `EthereumBlockWithTriggers` gone before merging, but I might be missing why that is a large lift.

Reply:
The trouble here is that the runtime still wants an `EthereumBlockWithTriggers`; going from the generic block stream to the concrete runtime requires turning a `C::Block` into an `EthereumBlockWithTriggers`, which right now is done through the awful `TriggerFilter.convert_block`. Once the runtime also accepts `C::Block`, we can pull all of `graph::components::ethereum::types` into `chain::ethereum`.

Reply:
Ah I see now. Maybe the `convert_block` could be done on `C::Block` then, but since this is temporary it's fine.